framework/src/util/LDAPExpr.cpp

```diff
@@ -709,14 +709,23 @@ std::string LDAPExpr::ParseState::getAttributeName()
 
 std::string LDAPExpr::ParseState::getAttributeValue()
 {
+  int num_parens = 0;
   std::string sb;
   bool exit = false;
   while (!exit) {
     Byte c = peek();
```

Comment:
peek looks at the next character in the string and returns that character or it will throw an out of range exception if m_pos is greater than the end of the string. It seems like the only way this while loop will exit is if a mismatched set of parentheses is detected or if an exception is thrown. That doesn't seem right. Am I missing something?

Reply:
peek() will throw, if it goes out of bounds (i.e. didn't find the last ")", where exit is set to true)
the code was originally designed to check for any ")" to exit; now it only exits when the last matching ")" is found.